runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -870,7 +877,15 @@ func s:DeleteCommands()
     if empty(s:k_map_saved)
       nunmap K
     else
-      call mapset('n', 0, s:k_map_saved)
+      " call mapset('n', 0, s:k_map_saved)
+      let mode = s:k_map_saved.mode !=# ' ' ? s:k_map_saved.mode : ''
```

Comment:
Why is this needed? I don't see `mode` being used anywhere here.

Reply:
Oh shoot, my bad, it supposed to be the first argument to `nvim_set_keymap`, because it takes empty string as `:map`

Reply:
I'd test how `nvim_set_keymap`  behaves if `s:k_map_saved` is "empty" (meaning `mapset` is called on an nonexisting mapping). Might be perfectly fine without extra acrobatics!

Reply:
Not sure if I understand what you mean. You mean when `s:k_map_saved` is an empty dictionary?

Reply:
Yep -- which is what you're testing here, right? (You already know the only possible mode is `n `.)

Reply:
I did this because when you're defining a mapping with `:map` and then restoring it with `:nmap` it kinda splits it into two:
```
:map K
n  K           * ...
ov K           * ...
```
While doing it this way -- checking the actual mode -- sets it back as it was:
```
:map K
   K           * ...
```
But I guess this is still the same thing, so maybe it is indeed unnecessary. And `:unmap K` still removes both mappings.

Reply:
And the check if dictionary was empty was already there, I just didn't touch it.

Reply:
Really? The diff says it's new ;)

Reply:
You mean the `if empty(s:k_map_saved)` line, 877? That's weird, to me it starts on `" call mapset('n', 0, s:k_map_saved)`, line 880

Reply:
Ah, sorry, I'm talking about `s:k_map_saved.mode !=# ' ' ? s:k_map_saved.mode : ''`

Reply:
Anyway, there is an actual benefit and this isn't exactly a hot path, so it's good :)

Reply:
Oh okay. So basically the issue was that `nvim_get_keymap` returns a space (`' '`) as `:map`, and `nvim_set_keymap` expects an empty string (`''`) for `:map`.